shell/edit/workload/mixins/workload.js

```diff
@@ -148,21 +149,48 @@ export default {
 
     if (!this.value.spec) {
       this.value.spec = {};
+      if (this.value.type === 'pod') {
```

Comment:
Why are we checking if the value is a pod? We also need to set container specs for the other workload types.

Reply:
When we creating Workload > Pod the spec is different. 

https://kubernetes.io/docs/concepts/workloads/pods/

<img width="666" alt="image" src="https://user-images.githubusercontent.com/1387263/187726062-5542ae0b-ba7d-4bee-a8da-2e7dee50a971.png">

So when reading we map to existing UI schema

https://github.com/rancher/dashboard/pull/6632/files#diff-ac776780d70ceae4eafb989d0dcd9a13815410a801c040143e9f60492d7aeb4cR152-R164

Then when saving map it back to what it suppose to be.

https://github.com/rancher/dashboard/pull/6632/files#diff-220a296a514947b845622eabcde760f57893b60964f1de32ed5c5bb1a18176fbR184-R199